src/main/java/eu/ha3/presencefootsteps/sound/acoustics/AcousticsPlayer.java

```diff
@@ -1,23 +1,24 @@
 package eu.ha3.presencefootsteps.sound.acoustics;
 
-import java.util.*;
-
 import com.google.common.base.Strings;
-
 import eu.ha3.presencefootsteps.PresenceFootsteps;
-import eu.ha3.presencefootsteps.sound.State;
 import eu.ha3.presencefootsteps.sound.Options;
+import eu.ha3.presencefootsteps.sound.State;
 import eu.ha3.presencefootsteps.sound.player.SoundPlayer;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
 import net.minecraft.entity.LivingEntity;
+import org.jetbrains.annotations.NotNull;
 
-public class AcousticsPlayer implements AcousticLibrary {
-
-    private final SoundPlayer player;
+import java.util.Map;
+import java.util.stream.Stream;
 
-    private final Map<String, Acoustic> acoustics = new HashMap<>();
+public record AcousticsPlayer(
+        @NotNull SoundPlayer player,
+        @NotNull Map<String, Acoustic> acoustics
```

Comment:
You can remove the @ NotNull annotations. I instead annotate things with @ Nullable if they may be null